apricot/local/service.go

```diff
@@ -135,12 +135,18 @@ func (s *Service) GetHostInventory(detector string) (hosts []string, err error)
 		log.WithError(err).Fatal("Error, could not retrieve host list.")
 		return []string{}, err
 	}
-	i := 0
-	hosts = make([]string, len(keys))
+	hostSet := make(map[string]bool, len(keys))
 	for _, key := range keys {
 		hostTrimmed := strings.TrimPrefix(key, keyPrefix)
 		hostname := strings.Split(hostTrimmed, "/")
-		hosts[i] = hostname[0]
+		if _, ok := hostSet[hostname[0]]; !ok {
+			hostSet[hostname[0]] = true
+		}
```

Comment:
Since this is a set we can assume a write is fast. Better to just write directly i.e. `hostSet[hostname[0]] = true`. Worst case you're gonna overwrite entries with identical ones all the time, I'd argue it's unlikely that this is more expensive than checking before every write.